app/uk/gov/hmrc/tradergoodsprofilesrouter/factories/AuditEventFactory.scala

```diff
@@ -0,0 +1,129 @@
+import uk.gov.hmrc.tradergoodsprofilesrouter.service.DateTimeService.DateTimeFormat
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.HeaderNames
+
+class AuditEventFactory @Inject() (
+  dateTimeService: DateTimeService
+) {
+
+  private val auditSource = "trader-goods-profiles-router"
+  private val auditType   = "ManageGoodsRecord"
+
+  def removeRecord(
+    eori: String,
+    recordId: String,
+    actorId: String,
+    requestedDateTime: String,
+    status: String,
+    statusCode: Int,
+    failureReason: Option[Seq[String]] = None
+  )(implicit hc: HeaderCarrier): ExtendedDataEvent = {
+
+    val auditDetails = AuditRemoveRecordDetails(
+      clientId = hc.headers(Seq(HeaderNames.ClientId)).head._2,
+      requestDateTime = requestedDateTime,
+      responseDateTime = dateTimeService.timestamp.asStringSeconds,
+      outcome = AuditOutcome(status, statusCode, failureReason),
+      request = AuditRemoveRecordRequest(eori, recordId, actorId)
+    )
+
+    ExtendedDataEvent(
+      auditSource = auditSource,
+      auditType = auditType,
+      tags = hc.toAuditTags(),
+      detail = Json.toJson(auditDetails)
+    )
+  }
+
+  def createRecord(
+    createRecordRequest: CreateRecordRequest,
+    requestedDateTime: String,
+    status: String,
+    statusCode: Int,
+    failureReason: Option[Seq[String]] = None,
+    createOrUpdateRecordResponse: Option[CreateOrUpdateRecordResponse] = None
+  )(implicit hc: HeaderCarrier): ExtendedDataEvent = {
+
+    val auditDetails = AuditCreateRecordDetails(
+      clientId = hc.headers(Seq(HeaderNames.ClientId)).head._2,
+      requestDateTime = requestedDateTime,
+      responseDateTime = dateTimeService.timestamp.asStringSeconds,
+      outcome = AuditOutcome(status, statusCode, failureReason),
+      request = createRecordRequest,
+      response = createOrUpdateRecordResponse
+    )
+
+    ExtendedDataEvent(
+      auditSource = auditSource,
+      auditType = auditType,
+      tags = hc.toAuditTags(),
+      detail = Json.toJson(auditDetails)
+    )
+  }
+
+  def updateRecord(
+    updateRecordRequest: UpdateRecordRequest,
+    requestedDateTime: String,
+    status: String,
+    statusCode: Int,
+    failureReason: Option[Seq[String]] = None,
+    createOrUpdateRecordResponse: Option[CreateOrUpdateRecordResponse] = None
+  )(implicit hc: HeaderCarrier): ExtendedDataEvent = {
+
+    val auditDetails = AuditUpdateRecordDetails(
+      clientId = hc.headers(Seq(HeaderNames.ClientId)).head._2,
+      requestDateTime = requestedDateTime,
+      responseDateTime = dateTimeService.timestamp.asStringSeconds,
+      outcome = AuditOutcome(status, statusCode, failureReason),
+      request = updateRecordRequest,
+      response = createOrUpdateRecordResponse
+    )
+
+    ExtendedDataEvent(
+      auditSource = auditSource,
+      auditType = auditType,
+      tags = hc.toAuditTags(),
+      detail = Json.toJson(auditDetails)
+    )
+  }
+}
+
+object AuditEventFactory {
+
+  case class AuditOutcome(
```

Comment:
There is already an Outcome case class, I think this can be deleted and use that instead